src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java

```diff
@@ -2052,6 +2053,40 @@ public void testClosedResultSet() throws Exception {
             }
         }
 
+        /**
+         * Tests reusing Statement after being automatically closed via closeOnCompletion().
+         *
+         * @throws Exception
+         */
+        @Test
+        public void testReuseAfterCloseOnCompletion() throws Exception {
+            try (Connection conn = getConnection(); Statement stmt = conn.createStatement()) {
+
+                // enable isCloseOnCompletion
+                try {
+                    stmt.closeOnCompletion();
+                } catch (Exception e) {
+                    fail(TestResource.getResource("R_unexpectedException") + e.getMessage());
+                }
+
+                SQLServerResultSet rs = (SQLServerResultSet) stmt.executeQuery("SELECT 1");
```

Comment:
use try with resource on rs

Reply:
It's been a while since I submitted this PR, but I have a vague recollection that this was intentional to ensure rs was not cleaned up before the subsequent stmt.executeQuery() runs. I feel like I changed the code in the original PR to using try with resources and couldn't get it to fail before... I can try again. I'll add a comment if it really needs to be done without the try with resources.

Reply:
Done.